python/lsst/daf/butler/datastores/posixDatastore.py

```diff
@@ -106,14 +107,17 @@ def setConfigRoot(cls, root, config, full):
                                   toUpdate={"root": root},
                                   toCopy=("cls", ("records", "table")))
 
-    def __init__(self, config, registry):
+    def __init__(self, config, registry, butlerRoot=None):
         super().__init__(config, registry)
         if "root" not in self.config:
             raise ValueError("No root directory specified in configuration")
-        self.root = self.config["root"]
+
+        # Support repository relocation in config
+        self.root = replaceRoot(self.config["root"], butlerRoot)
```

Comment:
Just for my own edification, this expands the root in the in-memory config instance into an absolute path with no `<root>` placeholder, but never affects on-disk configuration, right?

Reply:
Correct.